src/views/Graph.vue

```diff
@@ -796,33 +775,18 @@ export default {
     },
 
     /**
-     * Check if node is collapsed by family or ancestor
-     * If not collapsed return null
-     * If the node is collapsed by first parent return the parent name
-     * If the node is collapsed by an ancestor further up the tree return the name of the ancestor
-     * @property {String} nodeFirstParent - name of the first parent
-     * @returns {String | null} name of the collapsed parent/ancestor or null
+     * Check if node is collapsed by an ancestor family
+     *
+     * @param {string} id - ID of the task/family/namespace
+     * @returns {string=} name of the collapsed ancestor if any
      */
-    isNodeCollapsedByFamily (nodeFirstParent) {
-      // the nodes first parent is collapsed
-      const firstParent = this.collapseFamily.includes(nodeFirstParent)
-      // a family member up the tree is collapsed
-      const ancestor = this.allParentLookUp[nodeFirstParent].some(element => {
-        return this.collapseFamily.includes(element)
-      })
-      if (firstParent && !ancestor) {
-        // the node is collapsed by its first parent
-        return nodeFirstParent
-      } else if (ancestor) {
-        // the node is collapsed by an ancestor
-        for (let i = this.allParentLookUp[nodeFirstParent].length - 1; i >= 0; i--) {
-          if (this.collapseFamily.includes(this.allParentLookUp[nodeFirstParent][i])) {
-            return this.allParentLookUp[nodeFirstParent][i]
-          }
-        }
-      } else {
-        // the node is not collapsed
-        return null
+    getCollapsedAncestor (id) {
+      const { name } = this.cylcTree.$index[id].node.firstParent
+      if (name !== 'root') {
+        // TODO: could use Set.intersection if browser support is good enough?
+        return this.collapseFamily.find((family) => (
+          name === family || this.allParentLookUp.get(name).includes(family)
+        ))
```

Comment:
A non-iterative approach would be very worthwhile here as this gets called a lot as part of graphviz document building.